### PR TITLE
feat: add support for delegate agent stuck check.

### PR DESCRIPTION
**Pull request by assertion**
After add the `Microagents and Delegation`  #1238 ，the _is_stuck won't be used to check if the micro agent is stuck. 
This pr will solve the problem.

### CODE REVIEW

File: opendevin/controller/agent_controller.py [hunk 1]
```diff
@@ -292,32 +295,36 @@ async def step(self, i: int) -> bool:
     def get_state(self):
         return self.state
 
-    def _is_stuck(self):
+    def _is_stuck(self, is_delegate=False):
```
Comment (rbren) [inline opendevin/controller/agent_controller.py:298]: `delegate` is just another `AgentController`. So we should be able to just say:
```
if self.delagate._is_stuck:
  return True
# rest of state checking logic
```

Comment (assertion) [reply opendevin/controller/agent_controller.py]: Thanks, it's modified.

**Comment by assertion** [comment]
resolve the conflicts.

File: opendevin/controller/agent_controller.py [hunk 2]
```diff
@@ -239,7 +237,10 @@ async def step(self, i: int) -> bool:
                 obs: Observation = AgentDelegateObservation(content='', outputs=outputs)
                 await self.add_history(NullAction(), obs)
                 self.delegate = None
-                self.delegateAction = None
+            elif self.delegate._is_stuck():
+                logger.info('Loop detected in DelegateAgent, stopping task')
+                await self.delegate.set_agent_state_to(AgentState.ERROR)
+                raise Exception('Loop detected, stopping task.')
             return False
```
Comment (rbren) [inline opendevin/controller/agent_controller.py:244]: I _think_ we can just put
```
if self.delegate and self.delegate._is_stuck():
  return True
```
inside `_is_stuck`, and remove this part entirely. That would be ideal at least.

**Pull request closed without merging**